gtsfm/utils/io.py

```diff
@@ -517,28 +527,17 @@ def read_scene_data_from_colmap_format(
     """
     # Determine whether scene data is stored in a text (txt) or binary (bin) file format.
     if Path(data_dir, "images.txt").exists():
-        file_format = "txt"
+        file_format = ".txt"
     elif Path(data_dir, "images.bin").exists():
-        file_format = "bin"
+        file_format = ".bin"
     else:
         raise ValueError(
             f"Unknown file format, as neither `{data_dir}/images.txt` or `{data_dir}/images.bin` could be found."
         )
-
-    if file_format == "txt":
-        # TODO(johnwlambert): Consider unifying interfaces by using `colmap_io.read_model` for txt reading also.
-        points_fpath = f"{data_dir}/points3D.txt"
-        images_fpath = f"{data_dir}/images.txt"
-        cameras_fpath = f"{data_dir}/cameras.txt"
-        wTi_list, img_fnames = read_images_txt(images_fpath)
-        calibrations, img_dims = read_cameras_txt(cameras_fpath)
-        point_cloud, rgb = read_points_txt(points_fpath)
-
-    elif file_format == "bin":
-        cameras, images, points3d = colmap_io.read_model(path=data_dir, ext=".bin")
-        img_fnames, wTi_list, calibrations, _, point_cloud, rgb, img_dims = colmap2gtsfm(
-            cameras, images, points3d, load_sfmtracks=False
-        )
+    cameras, images, points3d = colmap_io.read_model(path=data_dir, ext=file_format)
+    img_fnames, wTi_list, calibrations, _, point_cloud, rgb, img_dims = colmap2gtsfm(
```

Comment:
can we put this change in a separate PR?

I don't think this new function is equivalent to the old one, since the other one provides images in a lexicographically sorted order: https://github.com/borglab/gtsfm/blob/master/gtsfm/utils/io.py#L378

the .bin format does not do that, and it will create issues for any retrieval

Reply:
`colmap2gtsfm` now lexigraphically sorts by the filenames as before